docs(contributing): add updates proposal (#7)
bf2fc6cc711aee1a0c2a/fleet-manager-go-template: CONTRIBUTING.md

done manually since the two repositories are different. You are free to choose/rely on whichever 
automation tool you have at your disposal e.g the combination of [git patch](https://git-scm.com/docs/git-format-patch) 
and [git apply](https://git-scm.com/docs/git-apply) etc. Code fix is done on a roll forward fashion.

### Notifying others about updates

#### From service specific code to template

In case of updates that are coming from a service specific fleet manager to the template, open an issue
in [ffm project](https://github.com/bf2fc6cc711aee1a0c2a/ffm-project/issues) stating if it is a bug 
fix/enhancement/ version updates etc.  It is advisable to open PR once there is concensus on the adoption of 
the enhancement. To speed up the decision time, you can start a [zulip thread](https://bf2.zulipchat.com/) or 
[google group thread](https://groups.google.com/g/factorized-fleet-manager) engaging the interested parties 
asking if they would like to adopt the proposal. Additionally, it is okay to open a draft PR, to drive the conversation
and enrich the understanding of the proposal.

#### From template to specific fleet managers

> Once a fix / enhancement has landed to the template, an email should be sent to [factorised fleet manager google group](https://groups.google.com/g/factorized-fleet-manager). Likewise, the notification has to be sent to [zulip chat](https://bf2.zulipchat.com/).

Reviewer: Maybe instead of each fix/enhancement applied, it should be when we decide to create a new version/release of the template ?. Otherwise we could generate lots of noise.

Author: Okay, I guess let's try this for now and we'll adjust when it becomes a lot of noises.

### Notifying other of updates

#### From service specific code to template

In case of updates that are coming from a service specific fleet manager to the template, open an issue
in [ffm project]() stating if it is a bug fix/ enhancement/ version updates etc. To avoid wasting precious 
time, a PR should only be opened once there is concensus on the adoption of the enhancement. To speed up 
the decision time, you can start a [zulip thread](https://bf2.zulipchat.com/) or [google group thread](https://groups.google.com/g/factorized-fleet-manager) engaging the interested parties asking if they would like to adopt the proposal. 

#### From template to specific fleet managers

Once a fix / enhancement has landed to the template, an email should be sent to [factorised fleet manager google group](https://groups.google.com/g/factorized-fleet-manager). Likewise, the notification has to be sent to [zulip chat](https://bf2.zulipchat.com/).

>NOTE: Prefer an approach/ a process where the templates and service specific fleet managers are to be updated regulary. 
This will make the rather "manual" update process less painful to work with by avoiding huge drifts which are likely to 

> cause a painful bing-bang update.

Reviewer: `big-bang`

Author: Good catch! Thanks, to think I went over this part several times ha ha

# Contributing

> ## Templates update

Author: @r00ta @danielezonca can you also have a look at this section? It will be good to have similar contribution guide/pattern across the two template repositories.

Every team using the template has shared responsibility and code access to this repository.
The template will be collectively contributed and versioned.

When a bug fix or a feature change is done on the Go template, a pull request on the Java version 
(preferred) or an issue will be opened to make sure we keep both versions at the same feature and 
maturity level.

There is an agreement from each fleet-manager team to update their fleet-manager code on the newest 

> template regularly. Code upates to/from the template onto/from a specific service is expected to be 

Reviewer 2: ```suggestion
template regularly. Code updates to/from the template onto/from a specific service are expected to be 
```

done manually since the two repositories are different. You are free to choose/rely on whichever 
automation tool you have at your disposal e.g the combination of [git patch](https://git-scm.com/docs/git-format-patch) 

> and [git apply](https://git-scm.com/docs/git-apply) etc. Code fix is done on a roll forward fashion.

Reviewer 3: Do we expect to change this with phase 2 and phase 3?

Author: Nothing is fixed and the guide can be improved on par needs basis as we progress. The current proposal is meant to get things rolling and we'll be revisiting in the future.

> ### Notifying other of updates

Reviewer 2: ```suggestion
### Notifying others about updates
```

#### From service specific code to template

In case of updates that are coming from a service specific fleet manager to the template, open an issue
in [ffm project](https://github.com/bf2fc6cc711aee1a0c2a/ffm-project/issues) stating if it is a bug 

> fix/enhancement/ version updates etc. To avoid wasting precious time, a PR should only be opened once there 

Author: @emmanuelbernard can you have another look please?

> is concensus on the adoption of the enhancement. To speed up the decision time, you can start a 

Reviewer 2: ```suggestion
is consensus on the adoption of the enhancement. To speed up the decision time, you can start a 
```

[zulip thread](https://bf2.zulipchat.com/) or [google group thread](https://groups.google.com/g/factorized-fleet-manager) 
engaging the interested parties asking if they would like to adopt the proposal. 

#### From template to specific fleet managers

Once a fix / enhancement has landed to the template, an email should be sent to [factorised fleet manager google group](https://groups.google.com/g/factorized-fleet-manager). Likewise, the notification has to be sent to [zulip chat](https://bf2.zulipchat.com/).

> >NOTE: Prefer an approach/ a process where the templates and service specific fleet managers are to be updated regulary. 

Reviewer 2: ```suggestion
>NOTE: A preferred approach/ process is to update the templates and service specific fleet managers regularly. 
```

This will make the rather "manual" update process less painful to work with by avoiding huge drifts which are likely to 
cause a painful big-bang update.

## Definition of Done

> * All acceptance criteria specified in JIRA are met

Author: The definition of Done needs revisiting. It is an area which is likely out of date with the way of doing things in the ffm project.

> * All acceptance criteria specified in JIRA are met

Author: Updated the whole definition of done. 
It was a left over of the one we have in kas-fleet-manager.
